packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_pane_controller.dart

```diff
@@ -95,19 +100,34 @@ class MemoryChartPaneController extends DisposableController
   ValueListenable<bool> get paused => _paused;
   final _paused = ValueNotifier<bool>(true);
   void pause() => _paused.value = true;
-  Future<void> resume() async {
-    if (!_paused.value) return;
-    if (mode != DevToolsMode.connected) throw StateError('Not connected.');
-    await _chartConnection!.maybeConnect();
+  void resume() {
+    assert(mode == DevToolsMode.connected && _chartConnection != null);
     _paused.value = false;
   }
 
+  /// Connects when chart is first time expanded.
```

Comment:
Ubernit: 

"Establishes the chart connection when the chart is expanded for the first time.

Returns true if the chart was already connected or the connection was established."